mlir/test/Conversion/SCFToAffine/scf-affine-round-trip.mlir

```diff
@@ -0,0 +1,36 @@
+// RUN: imex-opt %s --scf-to-affine -lower-affine  -allow-unregistered-dialect -split-input-file| FileCheck %s
```

Comment:
It'd be a stricter test if it wouldn't need "-allow-unregistered-dialect" (in other tests as well). Why is this needed?

Reply:
`-allow-unregistered-dialect` has nothing to do with 'strictness' and usually used to define ops on-the-fly in the tests. It wasn't used in these specific tests though.

Reply:
If the test has more freedom in what it can produce there is more room for undetected errors, e.g. false-positives.

Reply:
Probably yes, but it will be hard to construct such an example in practice where it both a valid test and susceptible to such a error and I've never saw such one.